modules/cluster/cloud-config.yml

```diff
@@ -1,112 +1,111 @@
+                "swap": {
+                    "measurement": [
+                        "swap_used_percent"
+                    ],
+                    "metrics_collection_interval": 5
+                },
+                "mem": {
+                    "measurement": [
+                        "mem_used_percent"
+                    ],
+                    "metrics_collection_interval": 5
+                }
+            },
+            "append_dimensions": {
+                "ImageId": "$${aws:ImageId}",
+                "InstanceId": "$${aws:InstanceId}",
+                "InstanceType": "$${aws:InstanceType}",
+                "AutoScalingGroupName": "$${aws:AutoScalingGroupName}"
+            },
+            "aggregation_dimensions": [["InstanceId"], ["AutoScalingGroupName"]]
+        },
+        "logs": {
+            "logs_collected": {
+                "files": {
+                    "collect_list": [
+                        {
+                            "file_path": "/opt/aws/amazon-cloudwatch-agent/logs/amazon-cloudwatch-agent.log*",
+                            "log_group_name": "${log_group_name}",
+                            "log_stream_name": "{instance_id}/amazon-cloudwatch-agent",
+                            "timezone": "UTC"
+                        },
+                        {
+                            "file_path": "/var/log/ecs/ecs-agent.log*",
+                            "log_group_name": "${log_group_name}",
+                            "log_stream_name": "{instance_id}/amazon-ecs-agent",
+                            "timezone": "UTC"
+                        }
+                    ]
+                }
+            },
+            "log_stream_name": "{instance_id}/untitled-logs"
+        }
+    }
 
-  - path: "/etc/ecs/ecs.config"
-    permissions: "0744"
-    owner: "root"
-    content: |
-      ECS_CLUSTER=${ecs_cluster_name}
-      ECS_LOG_LEVEL=${ecs_log_level}
-      ECS_ENABLE_CONTAINER_METADATA=true
-      ECS_ENABLE_TASK_IAM_ROLE=true
-      ECS_ENABLE_TASK_IAM_ROLE_NETWORK_HOST=true
-      ECS_AVAILABLE_LOGGING_DRIVERS=["awslogs"]
+- path: "/etc/ecs/ecs.config"
+  permissions: "0744"
+  owner: "root"
+  content: |
+    ECS_CLUSTER=${ecs_cluster_name}
+    ECS_LOG_LEVEL=${ecs_log_level}
+    ECS_ENABLE_CONTAINER_METADATA=true
+    ECS_ENABLE_TASK_IAM_ROLE=true
+    ECS_ENABLE_TASK_IAM_ROLE_NETWORK_HOST=true
+    ECS_AVAILABLE_LOGGING_DRIVERS=["awslogs"]
 
-  - path: "/usr/local/scripts/cloudformation-signal.sh"
-    permissions: "0744"
-    owner: "root"
-    content: |
-      #! /bin/bash
-
-      set -euo pipefail
-
-      function await_process() {
-        echo -n "Waiting for $1..."
-        while ! pgrep -f "$1" > /dev/null; do
-          sleep 1
-        done
-        echo "Done!"
-      }
-      await_process "/usr/libexec/amazon-ecs-init start"
+- path: "/usr/local/scripts/cloudformation-signal.sh"
+  permissions: "0744"
+  owner: "root"
+  content: |
+    #! /bin/bash
+    set -uo pipefail
```

Comment:
I'd really like us to have `-e` in here to avoid confusing issues 🤔

Reply:
An alternative is to use `set +e` (and then `set -e` again) inside the while loop, or `grep <stuff> || echo ""`  to return an empty string if grep does not find a match (which will continue the loop).